politeiawww/proposals.go

```diff
@@ -1504,25 +1633,78 @@ func voteResults(sv www.StartVote, cv []www.CastVote) []www.VoteOptionResult {
 	return results
 }
 
-// setVoteStatusReply stores the given VoteStatusReply in memory.  This is to
-// only be used for proposals whose voting period has ended so that we don't
-// have to worry about cache invalidation issues.
+// setVoteStatusReply converts a VoteStatusReply to a VoteSummary and stores it
+// in memory.  This is to only be used for proposals whose voting period has
+// ended so that we don't have to worry about cache invalidation issues.
 //
 // This function must be called without the lock held.
-func (p *politeiawww) setVoteStatusReply(v www.VoteStatusReply) {
+//
+// ** This fuction is to be removed when the deprecated vote status route is
+// ** removed.
+func (p *politeiawww) setVoteStatusReply(v www.VoteStatusReply) error {
 	p.Lock()
 	defer p.Unlock()
 
-	p.voteStatuses[v.Token] = v
+	endHeight, err := strconv.Atoi(v.EndHeight)
+	if err != nil {
+		return err
+	}
+
+	voteSummary := www.VoteSummary{
+		Status:           v.Status,
+		EligibleTickets:  uint32(v.NumOfEligibleVotes),
+		EndHeight:        uint64(endHeight),
+		QuorumPercentage: v.QuorumPercentage,
+		PassPercentage:   v.PassPercentage,
+		Results:          v.OptionsResult,
+	}
+
+	p.voteSummaries[v.Token] = voteSummary
+
+	return nil
 }
 
-func (p *politeiawww) voteStatusReply(token string, bestBlock uint64) (*www.VoteStatusReply, error) {
+// getVoteStatusReply retrieves the VoteSummary from the cache for a proposal
+// whose voting period has ended and converts it to a VoteStatusReply.
+//
+// This function must be called without the lock held.
+//
+// ** This fuction is to be removed when the deprecated vote status route is
+// ** removed.
+func (p *politeiawww) getVoteStatusReply(token string) (*www.VoteStatusReply, bool) {
 	p.RLock()
-	vsr, ok := p.voteStatuses[token]
+	vs, ok := p.voteSummaries[token]
 	p.RUnlock()
+
+	if !ok {
+		return nil, false
+	}
+
+	totalVotes := uint64(0)
```

Comment:
Style nit. If you're initializing a variable to its zero value just do:

```suggestion
	var totalVotes uint64
```